### PR TITLE
Fix clippy warnings for 1.83

### PR DESCRIPTION
**Pull request by GearsDatapacks**
This PR fixes newly occurring clippy warnings as of Rust v1.83

### CODE REVIEW

File: compiler-core/src/analyse.rs [hunk 1]
```diff
@@ -1762,10 +1762,10 @@ fn get_compatible_record_fields<A: std::fmt::Debug>(
             };
 
             // The labels must be the same
-            if !argument
+            if argument
                 .label
                 .as_ref()
-                .is_some_and(|(_, arg_label)| arg_label == first_label)
+                .is_none_or(|(_, arg_label)| arg_label != first_label)
```
Comment (lpil) [inline compiler-core/src/analyse.rs:1768]: this is so weird

**Review by lpil — APPROVED** [review]
Thank you!!

File: compiler-core/src/analyse.rs (hunk 1 above)
Comment (GearsDatapacks) [reply compiler-core/src/analyse.rs:1768]: I thought so too. We could disable this lint if we wanted

Comment (lpil) [reply compiler-core/src/analyse.rs:1768]: If it comes up a few times let's give it a think

**Pull request merged**